### PR TITLE
Unspecified `"scale"` in table headers defaults to `False`

### PR DESCRIPTION
**Pull request by vladsavelyev**
Creating a separate PR to fix these warnings ([from this issue](https://github.com/ewels/MultiQC/issues/2174#issuecomment-1809908750_)):

```
|         ngsderive | Found 2 reports
|        mqc_colour | ngsderive_instruments_plot: Error getting colour: could not convert string to float: 'e'
|        mqc_colour | ngsderive_instruments_plot: Error getting colour: could not convert string to float: 'e'
```

That's happening for columns with `None` or non-numeric values like this:

Screenshot 2023-11-14 at 12 24 28

If `"scale"` is not set, so it gets the default value `"GnBu"`

```py
headers["basis"] = {
    "title": "Instrument: Basis",
    "description": "Basis upon which the prediction was made.",
}
```

Which leads this `float` parse call of strings and `None` values to fail:
https://github.com/ewels/MultiQC/blob/master/multiqc/utils/mqc_colour.py#L391

As far as I understand, we want users to always explicitly set a scale, so in this case they may want to set `"scale": False` if they don't want a scale at all. So I believe this should be a lint error - so modifying the code accordingly.

Another approach might be to add some check not to require a scale if the values are non-numeric, and instead set it to `False` automatically instead of `"GnBu"`.

UPD:

There are so many places in the code where the scale is not set, so it would be more practical to default unset scale to `False`.

### CODE REVIEW

**Comment by ewels** [comment]
> we want users to always explicitly set a scale, so in this case they may want to set `"scale": False` if they don't want a scale at all. So I believe this should be a lint error - so modifying the code accordingly.

I like this solution 👍🏻

**Comment by vladsavelyev** [comment]
It's not really practical: we get so many warnings about unset scale (many of those from dynamically parsed column headers) that it looks like a lot of work to set scales everywhere explicitly 😢 . That's just 10% of that:

```
plicate (ivar_variants-1) ## self.add_section(name=section_name, anchor=c_id, description=section_description, plot=plot, content=content)
"scale" is not specified for the key "HiSeq 1500". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "HiSeq 2000". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "HiSeq 2500". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "phenotype". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "n_het". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "n_hom_ref". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "n_hom_alt". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "n_unknown". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "X_het". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "X_hom_ref". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "X_hom_alt". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "X_n". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "Y_n". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "Total Read Number". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "Alternative Alignments". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "Chimeric Reads". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "Duplicate Reads". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "End 1 Mapped Reads". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "End 2 Mapped Reads". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "End 1 Mismatches". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "End 2 Mismatches". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "End 1 Sense". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
"scale" is not specified for the key "End 2 Sense". Please set it explicitly to one of the possible values (see https://multiqc.info/docs/development/modules/#table-colour-scales), or set to `None` to disable colouring.
...
```

File: multiqc/modules/ngsderive/ngsderive.py [hunk 1]
```diff
@@ -211,6 +216,7 @@ def add_instrument_data(self):
             "instrument": {
                 "title": "Predicted Instrument",
                 "description": "Predicted instrument from ngsderive",
+                "scale": False,
```
Comment (vladsavelyev) [inline multiqc/modules/ngsderive/ngsderive.py:219]:
```suggestion
```

File: multiqc/modules/ngsderive/ngsderive.py [hunk 2]
```diff
@@ -244,6 +250,7 @@ def add_instrument_data(self):
         headers["basis"] = {
             "title": "Instrument: Basis",
             "description": "Basis upon which the prediction was made.",
+            "scale": False,
```
Comment (vladsavelyev) [inline multiqc/modules/ngsderive/ngsderive.py:253]:
```suggestion
```

File: multiqc/modules/ngsderive/ngsderive.py [hunk 3]
```diff
@@ -345,10 +352,12 @@ def add_encoding_data(self):
             "probable_encoding": {
                 "title": "Probable Encoding",
                 "description": "Predicted PHRED score encoding from ngsderive",
+                "scale": False,
```
Comment (vladsavelyev) [inline multiqc/modules/ngsderive/ngsderive.py:355]:
```suggestion
```

File: multiqc/modules/ngsderive/ngsderive.py [hunk 4]
```diff
@@ -345,10 +352,12 @@ def add_encoding_data(self):
...
             },
             "evidence": {
                 "title": "Encoding: Evidence",
                 "description": "Observed ASCII value ranges in PHRED score encoding",
+                "scale": False,
```
Comment (vladsavelyev) [inline multiqc/modules/ngsderive/ngsderive.py:360]:
```suggestion
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -19,6 +19,7 @@ Highlights:
 - Remove position:absolute from table values ([#2169](https://github.com/ewels/MultiQC/pull/2169))
 - Fix custom anchors for kraken ([#2170](https://github.com/ewels/MultiQC/pull/2170))
 - Fix logging spillover ([#2174](https://github.com/ewels/MultiQC/pull/2174))
+- Lint not setting `scale=False` for non-numeric values in table ([#2176](https://github.com/ewels/MultiQC/pull/2176))
```
Comment (vladsavelyev) [inline CHANGELOG.md:22]:
```suggestion
- Unspecified `"scale"` in table headers defaults to `False`([#2176](https://github.com/ewels/MultiQC/pull/2176))
```

**Comment by ewels** [comment]
If I do this diff:

```diff
diff --git a/multiqc/plots/table.py b/multiqc/plots/table.py
index c940d6a0..55f3e3f9 100644
--- a/multiqc/plots/table.py
+++ b/multiqc/plots/table.py
@@ -123,7 +123,7 @@ def make_table(dt: table_object.DataTable):
         )
 
         # Make a colour scale
-        if not header.get("scale"):
+        if header["scale"] is False:
             c_scale = None
         else:
             c_scale = mqc_colour.mqc_colour_scale(
diff --git a/multiqc/plots/table_object.py b/multiqc/plots/table_object.py
index e4c1bb47..247205c6 100644
--- a/multiqc/plots/table_object.py
+++ b/multiqc/plots/table_object.py
@@ -131,7 +131,7 @@ class DataTable:
                 headers[idx][k]["namespace"] = headers[idx][k].get("namespace", pconfig.get("namespace", ""))
                 headers[idx][k]["title"] = headers[idx][k].get("title", k)
                 headers[idx][k]["description"] = headers[idx][k].get("description", headers[idx][k]["title"])
-                headers[idx][k]["scale"] = headers[idx][k].get("scale", pconfig.get("scale", False))
+                headers[idx][k]["scale"] = headers[idx][k].get("scale", pconfig.get("scale", "GnBu"))
                 headers[idx][k]["format"] = headers[idx][k].get("format", pconfig.get("format", "{:,.1f}"))
                 headers[idx][k]["colour"] = headers[idx][k].get("colour", pconfig.get("colour", None))
                 headers[idx][k]["hidden"] = headers[idx][k].get("hidden", pconfig.get("hidden", None))

```

Then we don't lose any table formatting (tested with output from Somalier), however with the other improvements in the PR we also don't get any errors from tables with non-numeric values (tested with qc3c).

What do you think @vladsavelyev ?

File: multiqc/plots/table.py [hunk 6]
```diff
@@ -123,7 +123,7 @@ def make_table(dt: table_object.DataTable):
         )
 
         # Make a colour scale
-        if header["scale"] is False:
+        if not header.get("scale"):
```
Comment (ewels) [inline multiqc/plots/table.py:126]:
```suggestion
        if header["scale"] is False:
```

**Review by ewels — COMMENTED** [review]
Suggested reverts

**Comment by vladsavelyev** [comment]
Yeah, sounds good! 
Though I'd re-add a more explicit warning saying that the default "GnBu" was set because "scale" wasn't explicitly unspecified

**Comment by vladsavelyev** [comment]
Should fix this https://github.com/ewels/MultiQC/issues/2045

**Comment by ewels** [comment]
ok, handful of warnings left now - I think that they're all from this Dragen module table, which supplies zero config 🤦🏻‍♂️ 

https://github.com/ewels/MultiQC/blob/caa5dd12ba33e0b8bcc9a70dafcca622df9e92cd/multiqc/modules/dragen/trimmer_metrics.py#L42

**Comment by vladsavelyev** [comment]
Should be good now!

File: multiqc/modules/interop/interop.py [hunk 7]
```diff
@@ -208,13 +224,11 @@ def run_metrics_summary_table(data):
         headers = {
             "Yield": {
                 "rid": "summary_Yield",
-                "title": "{}p Yield".format(config.base_count_prefix),
-                "description": 'The number of bases sequenced ({} base pairs over all "usable cycles"'.format(
-                    config.base_count_desc
-                ),
+                "title": "Gbp Yield",
```
Comment (ewels) [inline multiqc/modules/interop/interop.py:227]:
```suggestion
                "title": "Gbp Yield",  # Numbers are rounded up to Gbp, so no point in using multiplier for smaller numbers as will be all zeroes
```

**Pull request merged**